Dockerfile

```diff
@@ -1,17 +1,25 @@
 FROM golang:alpine
 
 # Create application directory
-RUN mkdir /app
-ADD . /app/
-WORKDIR /app
+RUN mkdir /src
+ADD . /src/
+WORKDIR /src
+ENV CGO_ENABLED=0 GOOS=linux GOARCH=amd64
 
 # Build the application
-RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go mod download
-RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -o run .
+COPY go.* .
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 \
```

Comment:
You have already exported these as env variables. Why do you re-declare them here?

Reply:
For some reasons, not including them creates an error while building an image. Though I tried removing it just now and it seems to work.... somehow. I will remove that in the next commit and see if CI works.

Reply:
Oh, apparently without that it doesn't work on the `master` branch, but it works fine on the `sourcemod` branch which I am working on... I have no idea why, but here's the error log:

```
 => ERROR [stage-0 6/8] RUN --mount=type=cache,target=/go/pkg/mod     go mod graph | awk '{if ($1 !~ "@") print $2}' | xargs -r go get                                                       26.3s 
------
 > [stage-0 6/8] RUN --mount=type=cache,target=/go/pkg/mod     go mod graph | awk '{if ($1 !~ "@") print $2}' | xargs -r go get:
#11 22.02 go: downloading google.golang.org/protobuf v1.27.0
#11 22.02 go: downloading github.com/golang/protobuf v1.5.2
#11 22.02 go: downloading github.com/kelseyhightower/envconfig v1.4.0
#11 22.10 go: downloading github.com/prometheus/client_golang v1.11.0
#11 22.10 go: downloading github.com/stretchr/testify v1.5.1
#11 22.10 go: downloading github.com/gorilla/mux v1.8.0
#11 22.10 go: downloading github.com/prometheus/common v0.29.0
#11 22.10 go: downloading golang.org/x/sys v0.0.0-20210616094352-59db8d763f22
#11 22.10 go: downloading github.com/patrickmn/go-cache v2.1.0+incompatible
#11 22.10 go: downloading github.com/gorilla/websocket v1.4.2
#11 23.61 go: downloading github.com/davecgh/go-spew v1.1.1
#11 23.61 go: downloading github.com/pmezard/go-difflib v1.0.0
#11 23.61 go: downloading github.com/stretchr/objx v0.1.1
#11 23.62 go: downloading gopkg.in/yaml.v2 v2.4.0
#11 23.97 google.golang.org/protobuf: no Go source files
#11 24.02 go: downloading github.com/patrickmn/go-cache v1.0.0
#11 25.22 package google.golang.org/protobuf: build constraints exclude all Go files in /go/pkg/mod/google.golang.org/protobuf@v1.27.0
```

Reply:
So apparently the ``protobuf`` package is the issue, as it is not present in the ``sourcemod`` branch. It is commented "indirect" in `go.mod`, not sure what that means. Or maybe the issue is env variables is not working properly.

Reply:
The `// indirect` comment tells commands like `go mod tidy` to not remove the dependency, even if it seems to be unused. That is useful in case you have a dependecy that changes your programm, even if it not directly called within your code.